optimum/intel/ipex/modeling_base.py

```diff
@@ -652,11 +679,22 @@ def _prepare_generation_config(
         return generation_config, model_kwargs
 
     def generate(self, *args, **kwargs):
-        if self._is_ipex_exported and kwargs.get("assistant_model", None):
+        if is_ipex_version("<", "2.4.0") and self._is_ipex_exported and kwargs.get("assistant_model", None):
             raise ValueError(
-                f"Assisted decoding is not supported for patched models for now, support methods are {_IPEX_EXPORTED_GENERATION_METHODS}"
+                f"Assisted decoding is not supported for patched models if ipex < 2.4, support methods are {_IPEX_EXPORTED_GENERATION_METHODS}"
             )
-        return super().generate(*args, **kwargs)
+        # Patch functions to support IAKV cache
+        if self._is_ipex_exported:
+            _patch_crop_past_key_values()
+
+        result = super().generate(*args, **kwargs)
+
+        return result
+
+
+def _patch_crop_past_key_values():
+    transformers.generation.candidate_generator._crop_past_key_values = _ipex_crop_past_key_values
+    transformers.generation.utils._crop_past_key_values = _ipex_crop_past_key_values
```

Comment:
could this be unpatch after export ?

Reply:
As we discussed in Teams, this is the only way to enable all assisted decoding cases:
1. transformers target model + ipex draft model
2. ipex target model + transformers draft model
3. ipex target model + ipex draft model

Cause the `_crop_past_key_values` function is the same level as the model, we cannot do un-patch inside the generate function because it will run after generate, see [here](https://github.com/huggingface/transformers/blob/main/src/transformers/generation/candidate_generator.py#L209-L227). 

I have checked the model type inside the `_ipex_crop_past_key_values`. It only has impact on IPEX model, transformers model will go into the original function, so there is no risk even we don't unpatch.